cdap-ui/app/cdap/components/CaskWizards/AddNamespace/index.js

```diff
@@ -36,8 +37,14 @@ export default class AddNamespaceWizard extends Component {
   createNamespace() {
     return PublishNamespace()
       .flatMap(
-        () => {
-          return PublishPreferences();
+        (res) => {
+          if (res.includes('already exists')) {
+            return Rx.Observable.create((observer) => {
```

Comment:
I think if you just need to throw an error, you can use `Rx.Observable.throw`.

https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/throw.md